### PR TITLE
add more tests for app-sync.js

### PR DESCRIPTION
**Pull request by tcrasset**
~Needs #423 to be merged first~

### CODE REVIEW

**Comment by MatissJanis** [comment]
Happy to review once conflicts are solved

**Comment by tcrasset** [comment]
> Happy to review once conflicts are solved

Done :)

File: src/app-sync.js [hunk 1]
```diff
@@ -12,7 +12,9 @@ import { SyncProtoBuf } from '@actual-app/crdt';
 
 const app = express();
 app.use(errorMiddleware);
+app.use(express.json());
 app.use(express.raw({ type: 'application/actual-sync' }));
+app.use(express.raw({ type: 'application/encrypted-file' }));
```
Comment (MatissJanis) [inline src/app-sync.js:17]: Mind explaining this header? Why do we need it?

Comment (tcrasset) [reply src/app-sync.js:17]: It's used on the `/upload-user-file` endpoint.


Source: https://github.com/actualbudget/actual/blob/d6afc85a8c179d0e4c6ac3249097a4d1989f9c60/packages/loot-core/src/server/cloud-storage.ts#L278

I alluded to it in one of my previous PRs, but the middlewares here are needed when running tests to make sure that the request/response bodies are parsed correctly, but on production they work without it. I'm not fully sure why, but with the middlewares, it works in the test, and outside.

**Review by MatissJanis — APPROVED** [review]

**Comment by MatissJanis** [comment]
Thanks!

**Pull request merged**